src/index.cpp

```diff
@@ -1045,6 +1046,10 @@ namespace diskann {
     size_t pos = 0;
     for (auto it : best_L_nodes) {
       indices[pos] = it.id;
+      if (distances)
+      distances[pos] = it.distance;
+      if (_metric == diskann::INNER_PRODUCT)
+        distances[pos] = -distances[pos];
```

Comment:
This should be in an #ifndef EXEC_ENV_OLS block, because DLVS uses negdotprod as the metric.